### DISCUSSION TITLE
Questions about the mini-app chunks

### DISCUSSION THREAD

**Discussion by ra1nj** [callstack/repack#801] [Q&A] [answered]
I have a miniapp, it used a lot of node_modules libraries shared by host app, after i run the bundle common for the miniapp , i found that the third party node_modules libraries were also bundled, and they were been split into many chunks:
![Screenshot 2024-11-22 at 15 08 20](https://github.com/user-attachments/assets/feb62d39-e919-46f9-8c42-3cac1802d159)
To my understanding, i have configured those libraries as shared dependencies, those should not be packed in my miniapp's bundle/chunks, those should be packed in the host app's bundle.
![Screenshot 2024-11-22 at 15 21 22](https://github.com/user-attachments/assets/03b1ee23-f475-44d9-83cc-7b477ce678c5)

**Comment by jbroma** [comment, marked as answer]
the gist of it is that these modules are being emitted even when declared as shared deps but will never be used (as in downloaded from the mini app assets) - this can be turned off by adding `import: false` to each shared dependency configuration, you can read more about that here: https://webpack.js.org/plugins/module-federation-plugin/#import

**Reply by ra1nj** [reply]
I see.
